perllib/FixMyStreet/App/Form/Waste/Garden.pm

```diff
@@ -22,9 +22,18 @@ sub details_update_fields {
 
     my $cost_pa = $bin_count == 0 ? 0 : $form->{c}->cobrand->garden_waste_cost_pa($bin_count);
     my $cost_now_admin = $form->{c}->cobrand->garden_waste_new_bin_admin_fee($new_bins);
-    $form->{c}->stash->{cost_pa} = $cost_pa / 100;
-    $form->{c}->stash->{cost_now_admin} = $cost_now_admin / 100;
-    $form->{c}->stash->{cost_now} = ($cost_now_admin + $cost_pa) / 100;
+    if ($data->{apply_discount} && $c->stash->{waste_features}->{ggw_discount_pay_proportion}) {
+        my $proportion_to_pay = $c->stash->{waste_features}->{ggw_discount_pay_proportion};
+        $c->stash->{cost_pa} = (($cost_pa / 100) / 100) * $proportion_to_pay;
+        $c->stash->{per_bin_cost} = ($c->stash->{per_bin_cost} / 100) * $proportion_to_pay;
```

Comment:
Should this also change per_new_bin_cost/per_new_bin_first_cost (I realise neither are used for Brent, but in case this then gets used somewhere else)